src/screens/SendConfirmScreen.js

```diff
@@ -79,12 +80,17 @@ class SendConfirmScreen extends React.Component {
    *
    * @param {String} pin User PIN already validated
    */
-  executeSend = (pin) => {
+  executeSend = async (pin) => {
     const outputs = [{ address: this.address, value: this.amount, token: this.token.uid }];
     let sendTransaction;
 
     if (this.props.useWalletService) {
-      sendTransaction = new hathorLib.SendTransactionWalletService(this.props.wallet, { outputs });
+      await this.props.wallet.validateAndRenewAuthToken(pin);
```

Comment:
We probably don't need to wait as we're just updating the token, which is still valid.

Reply:
The problem here is that it might be expired as we don't schedule token renewals, we only check it when we are about to use it.

So if the user left the app open for 30 minutes and tries to create a transaction, it will ask for his pin twice: When signing inputs and on the renew token method

That's why I added the await there

Reply:
Got it. Can we do a local check if the token is still valid before making this call?

Reply:
I don't like having that much logic in the app. Maybe we could create a method `renewIfNeeded` or something like that in the lib

Reply:
@pedroferreira1 

Currently `validateAndRenewAuthToken` is only issuing a new token when the current one is either expired or will expire in less than 60s, so it is already a "renew if needed"

I guess that what @obiyankenobi is suggesting is to renew the token every time the user inserts his PIN on the app to reduce the number of times we ask for user input

Reply:
As far as I understood you are awaiting here because you don't know if you need to renew it or not and @obiyankenobi suggested to add a local check to await only in case it's needed. This local check that I don't like having in the app and I think should be in the lib. Not sure if you added it already but that's what I meant.

Reply:
Correct. Moreover, I suggest we renew the token (non blocking) every time the user inputs the pin.

Reply:
@pedroferreira1 @obiyankenobi 

`validateAndRenewAuthToken` will resolve without going to the wallet service if the token is still valid, so I don't see any reason to add this additional local check

In regards to renewing the token every time the user inputs the PIN, I agree and I will refactor the `validateAndRenewAuthToken` method to do this automatically (and non-blocking) everytime it receives a user-inputted PIN

This refactor will be done on the https://github.com/HathorNetwork/hathor-wallet-lib/pull/307 PR as it will be transparent for the wallet mobile

Reply:
@obiyankenobi @pedroferreira1 

Changes made on this line:
https://github.com/HathorNetwork/hathor-wallet-lib/pull/307/files#diff-6fa02af64d4fc88536c6977812dc0ee699b0d49377e7e85e7a5df5e274876bb9R634